extensions/eo/json-schema/schema.json

```diff
@@ -3,12 +3,26 @@
   "$id": "schema.json#",
   "title": "EO Extension",
   "description": "STAC EO Extension to a STAC Item.",
-  "allOf": [
+  "oneOf": [
     {
-      "$ref": "../../../item-spec/json-schema/item.json"
+      "allOf": [
+        {
+          "$ref": "../../../item-spec/json-schema/item.json"
+        },
+        {
+          "$ref": "#/definitions/eo"
```

Comment:
I don't get how this schema works. Why are Item and Collection refer both to the same eo definition?

Reply:
I have no idea - I was using another json-schema to figure out how define it and very likely got it all wrong.

Reply:
I can give it a try?!

Reply:
Well, let's wait until we sort out what we're doing with eo:bands and cloud_cover then come back to this